[CS2113-F10-3] TARBS

edemirkirkan opened the pull request.

Commit:
Set theme jekyll-theme-minimal

Commit:
Update AboutUs.md

chintaiann committed:
update AboutUs

bbawj committed:
Add brendan to aboutus

chintaiann committed:
Merge branch 'master' into taiann-AboutUs

chintaiann committed:
Merge pull request #1 from chintaiann/taiann-AboutUs

update AboutUs

timchang27 committed:
update about-us

bbawj committed:
Merge branch 'master' into master

bbawj committed:
Merge pull request #2 from bbawj/master

Add brendan to aboutus

chintaiann committed:
v1.0

Implement TravelPackage class
Adding of travel package
Removal of travel package through ID
Printing of all travel packages

todo: error handling

chintaiann committed:
Merge pull request #4 from chintaiann/master

v1.0

bbawj committed:
Add OOP abstractions

bbawj committed:
Merge pull request #5 from bbawj/master

Add OOP abstractions

timchang27 committed:
v1.0 - Some bugs

timchang27 committed:
v1.0 - Some bugs

timchang27 committed:
v1.0 - Some bugs

timchang27 committed:
v1.0 - Some bugs

chintaiann committed:
v1.0

1. added Reservation class
2. added Reservation arraylist to packages
3. ErrorCommand and ReservationCommand added

chintaiann committed:
Merge pull request #6 from chintaiann/master

v1.0

Commit:
make general revision

Commit:
revise the style

Commit:
revise the style

Commit:
Merge pull request #7 from edemirkirkan/master

make general revision

mafpovbul committed:
Added storage

timchang27 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-3/tp

 Conflicts:
	docs/AboutUs.md
	src/main/java/seedu/duke/Duke.java
	src/main/java/seedu/duke/Packages.java
	src/main/java/seedu/duke/Parser.java
	src/main/java/seedu/duke/Reservation.java
	src/main/java/seedu/duke/TravelPackage.java
	src/main/java/seedu/duke/command/AddCommand.java
	src/main/java/seedu/duke/command/ByeCommand.java
	src/main/java/seedu/duke/command/Command.java
	src/main/java/seedu/duke/command/DeleteCommand.java
	src/main/java/seedu/duke/command/PackagesCommand.java
	src/main/java/seedu/duke/command/ReservationCommand.java

timchang27 committed:
v1.0 - Initial Changes

Commit:
Added the 'help' command

Commit:
Revise the import issue

Commit:
Merge pull request #10 from edemirkirkan/master

Added help command and exception control mechanism

timchang27 committed:
Update DeveloperGuide.md

Initial Commit of Product Scope, User Stories, NFRs, and User Instructions

mafpovbul committed:
Merge pull request #8 from mafpovbul/master

Added storage

mafpovbul committed:
Refactored storage, added more exception handling

mafpovbul committed:
changed save file

mafpovbul committed:
Merge branch 'master' into master

mafpovbul committed:
Update DeveloperGuide.md

mafpovbul committed:
Merge pull request #11 from mafpovbul/master

Refactored storage, added more exception handling

mafpovbul committed:
Add files via upload

mafpovbul committed:
Update DeveloperGuide.md

mafpovbul committed:
Delete storage sequence diagram.png

mafpovbul committed:
Add files via upload

mafpovbul committed:
Update DeveloperGuide.md

mafpovbul committed:
Update DeveloperGuide.md

mafpovbul committed:
Update DeveloperGuide.md

mafpovbul committed:
Update UserGuide.md

mafpovbul committed:
Update UserGuide.md

bbawj committed:
Abstract reservations out from packages

bbawj committed:
Merge pull request #12 from bbawj/master

Abstract reservations out from packages

bbawj committed:
Update DeveloperGuide.md

bbawj committed:
Add ParserTest

bbawj committed:
Merge pull request #13 from bbawj/master

Add ParserTest

bbawj committed:
Update to TARBS

bbawj committed:
Merge pull request #14 from bbawj/master

Update to TARBS

mafpovbul committed:
Update DeveloperGuide.md

mafpovbul committed:
Delete StorageSeqDiag.png

mafpovbul committed:
Add files via upload

mafpovbul committed:
Delete StorageSeqDiag.png

mafpovbul committed:
Add files via upload

bbawj committed:
Update to TARBS

bbawj committed:
Merge pull request #15 from bbawj/master

Update to TARBS

bbawj committed:
Update DeveloperGuide.md

timchang27 committed:
Merge branch 'master' into master

timchang27 committed:
Merge pull request #9 from timchang27/master

Update About-us

chintaiann committed:
v2.0

1. all input standardized to using [COMMAND] [INPUT,INPUT,INPUT]
 -> using commas to separate between multiple inputs

2. reservations only used by TravelPackage

3. feature to add and delete reservations by providing travelpackageID and other details

4. commented out storage code, needs review after this commit

chintaiann committed:
Merge pull request #16 from chintaiann/master

v2.0

bbawj committed:
Add exception handling and print reservations

bbawj committed:
Merge pull request #17 from bbawj/master

Add exception handling and print reservations

a1021492980 commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +7,97 @@
+## Feature - Storage
+#### Initialisation (Loading Data)
+The sequence diagram that shows how `Storage` is created and the data is loaded from the saved files when the program is initialised is shown below: </br>
+![](StorageSeqDiag.png)
```
Perhaps can add an activation bar when you call the constructor of Storage?
<img width="242" alt="image" src="https://user-images.githubusercontent.com/60375639/160957918-2950b328-9f6a-4754-a3bc-51118d1b6668.png">

thedarie replied:
![image](https://user-images.githubusercontent.com/74888935/160958377-b8580c35-8730-43f8-9238-2b2fc570c3ea.png)
Should this line be a solid line since dotted lines are used for return values? Also, there should be an activation bar to represent the constructor and the entity storage is missing a colon

thedarie replied:
![image](https://user-images.githubusercontent.com/74888935/160958651-dddd42f1-81a2-449e-ac1b-089ec462be2f.png)
Since `parseReservationFile()` and `parseTravelPackagesFile()` are self-invocations, maybe adding 'overlapping' activation bars would be good

a1021492980 replied:
Perhaps can add colons for the class entity?

aiman-imtiaz replied:
The self-invocation calls should probably show small activation bars of their own, and should have return lines if something is being returned from the method calls.

<img width="251" alt="Screenshot 2022-03-31 at 11 22 16 AM" src="https://user-images.githubusercontent.com/49554923/160969945-9b3176fe-52b5-4921-ad87-f8f73290ac40.png">

aiman-imtiaz replied:
- Since this is an instance of the `Storage` class, it should be written as :Storage
- There should be a constructor shown that would instantiate an object of the `Storage` class
- The call to the constructor should have a return line that would return the object
- The call to the constructor should be represented by a solid line

<img width="223" alt="Screenshot 2022-03-31 at 11 27 46 AM" src="https://user-images.githubusercontent.com/49554923/160970492-5afd664b-cab9-4573-961e-1086f9910a45.png">

a1021492980 commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +7,97 @@
+
+## Feature - Storage
+#### Initialisation (Loading Data)
+The sequence diagram that shows how `Storage` is created and the data is loaded from the saved files when the program is initialised is shown below: </br>
```
Perhaps you mean
```suggestion
The sequence diagram that shows how `Storage` is created and the data is loaded from the saved files when the program is initialised is shown below: <br>
```

a1021492980 commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +7,97 @@
+
+For simplicity, `ByeCommand` is shown here:
+
+![image](https://user-images.githubusercontent.com/53790951/160289728-5a17b2ea-0599-48f4-a18b-cadfbbcbfb11.png)
```
Perhaps can add an activation bar when you call the constructor of ByeCommand?
<img width="117" alt="image" src="https://user-images.githubusercontent.com/60375639/160957869-efcc6863-2bdf-4a0c-9125-eebb65dee75f.png">

a1021492980 replied:
Is it better to include User in the left most of the sequence diagram?
<img width="608" alt="image" src="https://user-images.githubusercontent.com/60375639/160958336-a8b78376-cf49-4f5e-9feb-0fb152f339ad.png">

thedarie replied:
![image](https://user-images.githubusercontent.com/74888935/160959015-bb5dba17-de1b-4f3a-97ed-4ccb31c248e1.png)
Similar to the storage diagram, maybe the highlighted line should be a solid line

danbaterisna replied:
Should this vertical bar extending below the boundary of the `alt` box be there?

Also, I think this diagram would be clearer if it came with a step-by-step explanation.

![image](https://user-images.githubusercontent.com/8340533/160965833-5cc61d32-787c-4cf0-8148-cb79b13885f1.png)

aiman-imtiaz replied:
There should be a constructor activation bar here that should also return the instantiated object.

<img width="303" alt="Screenshot 2022-03-31 at 11 42 04 AM" src="https://user-images.githubusercontent.com/49554923/160971960-3a6bcaa2-46c2-4699-8a68-55e5ebf895e6.png">

a1021492980 reviewed (commented):
Good write up, just some minor suggestions for your consideration.

danbaterisna commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +7,97 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Basic Class Diagram 
+![image](https://user-images.githubusercontent.com/53790951/160271319-4a351f51-afd7-4e04-9451-f04143146551.png)
```
I think more descriptive roles can be used to describe these relationships, since things like `uses` or `instantiates` can already be implied by the arrows? Also, if `instantiates` implies an association, consider using a solid line for it instead.

![image](https://user-images.githubusercontent.com/8340533/160965280-da7e23ba-e500-4c69-bf67-a89075e2e903.png)

aiman-imtiaz replied:
Even if the class methods and fields are not represented, there should be another line separating the contents of the class.

<img width="220" alt="Screenshot 2022-03-31 at 11 39 38 AM" src="https://user-images.githubusercontent.com/49554923/160971756-9cabc8a5-0db1-4731-9581-c1d2080acbc1.png">

danbaterisna reviewed (commented):
Good explanation of existing features, can be improved with some changes to the diagrams

bbawj committed:
Add override equals for testing

bbawj committed:
Merge pull request #18 from bbawj/master

Add override equals for testing

mafpovbul committed:
Update DeveloperGuide.md

mafpovbul committed:
Add files via upload

mafpovbul committed:
Update DeveloperGuide.md

chintaiann committed:
v2.0

1. implemented functionality of not allowing reservations to be added if exceeds maxParticipants

2. error message when travelID not found during adding/removing reservations

3. adding/deletion of reservations will correctly alter number of participants now

chintaiann committed:
Merge pull request #19 from chintaiann/master

v2.0

chintaiann committed:
Update DeveloperGuide.md

chintaiann committed:
Update DeveloperGuide.md

mafpovbul committed:
Update storage

mafpovbul committed:
Merge pull request #20 from mafpovbul/master

Update storage

mafpovbul committed:
Update TARBS.java

mafpovbul committed:
Delete SaveFileSeqDiag.png

mafpovbul committed:
Delete StorageSeqDiag.png

mafpovbul committed:
Add files via upload

mafpovbul committed:
Delete SaveFileSeqDiag.png

mafpovbul committed:
Add files via upload

mafpovbul committed:
Update DeveloperGuide.md

bbawj committed:
Add SLAP for parser

bbawj committed:
Merge pull request #21 from bbawj/master

Add SLAP for parser

bbawj committed:
Update UserGuide.md

timchang27 committed:
v2.0 - Edited packages command and added info command

timchang27 committed:
Merge pull request #22 from timchang27/master

v2.0 - Edited packages command and added info command

timchang27 committed:
Update DeveloperGuide.md

timchang27 committed:
v2.0 - Edited packages command and added info command

timchang27 committed:
Merge pull request #23 from timchang27/master

v2.0 - Edited packages command and added info command

mafpovbul committed:
Exception handling

mafpovbul committed:
Merge pull request #24 from mafpovbul/master

Exception handling

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +8,127 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Basic Class Diagram 
+![image](https://user-images.githubusercontent.com/53790951/160271319-4a351f51-afd7-4e04-9451-f04143146551.png)
```
Remember to use standard UML syntaxes taught in the module. Some of the notations are not compliant with the notation covered in the module. 
Consider adding multiplicity to and other detailed information to the diagram to help understanding.

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +8,127 @@
+## Feature - Storage
+#### Initialisation (Loading Data)
+The sequence diagram that shows how `Storage` is created and the data is loaded from the saved files when the program is initialised is shown below: </br>
+![](LoadFileSeqDiag.png)
```
Consider using a consistent UML diagramming software unless the primary has limitations.

xseh replied:
1. Missing colon in front of the classes. Refer to the handout for more information on the syntax. 
2. Missing activation bats for calling getPackage(), saveTravelPackage(), instantiating FileWriter and write (text)*.
4. Consider rephrasing the condition for the loop (until all elements are iterated)
5. Consider adding cross to indicate EoL, if any. 

* If you would like to omit the information please ensure that there would not be a loss of data (e.g. can others tell that FileWriter was instantiated in Storage or before?)

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +8,127 @@
+#### Saving data
+The sequence diagram that shows how `Storage` is used to save the current list of travel packages and reservations is shown below: </br>
+
+![](SaveFilesSeqDiag.png)
```
Same as previous comment

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +8,127 @@
+The sequence diagram that shows how `Storage` is created and the data is loaded from the saved files when the program is initialised is shown below: </br>
+![](LoadFileSeqDiag.png)
+
+1. `TARBS` creates a Storage object with the relevant file name (`filePath`).
```
Instantiation of an object should involve the use of a constructor, which has activation bars connected to the Storage. Compare: 
![image](https://user-images.githubusercontent.com/39018497/161184837-4238c14f-0331-4bd2-b58d-a25c3f7cdce4.png)
and 
![image](https://user-images.githubusercontent.com/39018497/161185058-3ec5eede-f380-4f92-a875-9ef88dcea4a9.png)

xseh commented on docs/DeveloperGuide.md:
```diff
@@ -7,32 +8,127 @@
+
+For simplicity, `ByeCommand` is shown here:
+
+![image](https://user-images.githubusercontent.com/53790951/160289728-5a17b2ea-0599-48f4-a18b-cadfbbcbfb11.png)
```
1. Consider moving the condition for the [alt] path to beside the icon, instead of on the first method call. 
2. Consider adding crosses to indicate EoL if applicable
3. The classes are lacking a colon
4. Is there any code after return ByeCommand? If not consider ending both activation bars on the return arrow.

xseh reviewed (commented):
Overall good attempt! Do be more consistent with the syntaxes used in the module.

bbawj committed:
Add Date support

bbawj committed:
Merge pull request #25 from bbawj/master

Add Date support

bbawj committed:
Add .jar to gitignore

mafpovbul committed:
Exception handling for storage

mafpovbul committed:
Merge pull request #79 from mafpovbul/master

Exception handling for storage

bbawj committed:
Merge remote-tracking branch 'upstream/master'

bbawj committed:
Update help command, closes #77, closes #73, closes #51

bbawj committed:
Merge pull request #80 from bbawj/master

Update help command

bbawj committed:
Code review closes #26

bbawj committed:
Merge pull request #81 from bbawj/master

Code review closes #26

mafpovbul committed:
Date exception handling

mafpovbul committed:
Date exception handling

mafpovbul committed:
Merge pull request #82 from mafpovbul/master

Date exception handling

bbawj committed:
Add exception handling and tests, closes #42, closes #69, closes #66

mafpovbul committed:
Find package logic, added error messages

mafpovbul committed:
Merge pull request #83 from mafpovbul/master

Find package logic, added error messages

bbawj committed:
Fix conflict

bbawj committed:
Merge pull request #84 from bbawj/master

Date and price exception handling

mafpovbul committed:
Error checking when loading storage data

mafpovbul committed:
Merge pull request #85 from mafpovbul/master

Error checking when loading storage data

timchang27 committed:
v2.0 - Removed Unused Command Classes and Deleted Dead Code

timchang27 committed:
Merge pull request #86 from timchang27/master

v2.0 - Removed Unused Command Classes and Deleted Dead Code

mafpovbul committed:
Add storage test code

mafpovbul committed:
Merge pull request #87 from mafpovbul/master

Add storage test code

bbawj committed:
Add date range test

bbawj committed:
Merge pull request #88 from bbawj/master

Add date range test

chintaiann committed:
small fix

closes #27

chintaiann committed:
Merge pull request #89 from chintaiann/master

small fix

chintaiann committed:
Update chintaiann.md

mafpovbul committed:
Create justyn.md

mafpovbul committed:
Update justyn.md

bbawj committed:
Add Brendan PPP

bbawj committed:
Merge pull request #90 from bbawj/master

Add Brendan PPP

Commit:
Merge pull request #1 from AY2122S2-CS2113-F10-3/master

Update

Commit:
Fix the same package bug

Commit:
Fix the big integer delete bug

timchang27 committed:
Update Project Portfolio Page

Commit:
Fix exit command documentation bug

Commit:
Update user guide and help documentation

Commit:
Fix documentation issue

Commit:
Update documentation

timchang27 committed:
Resolve Issue

timchang27 committed:
Merge pull request #91 from timchang27/master

Update PPP and Resolve 1 Issue

timchang27 committed:
Improved Help and HelpCommand, Corrected Add Instructions

timchang27 committed:
Format Data

timchang27 committed:
Merge pull request #92 from timchang27/master

Improved Help and HelpCommand

timchang27 committed:
Update UserGuide.md

timchang27 committed:
Added PPP

timchang27 committed:
Update UserGuide.md

Commit:
Merge branch 'master' into master

Commit:
Merge pull request #93 from edemirkirkan/master

Update

bbawj committed:
Rename brendan PPP

mafpovbul committed:
Update justyn.md

bbawj committed:
Merge pull request #94 from bbawj/master

Rename brendan PPP

mafpovbul committed:
Rename justyn.md to mafpovbul.md

timchang27 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-3/tp

timchang27 committed:
Rename timothy.md to timchang27.md

timchang27 committed:
Merge pull request #95 from timchang27/master

Rename timothy.md to timchang27.md

timchang27 committed:
Added Hotel and Country Class

timchang27 committed:
Merge pull request #96 from timchang27/master

Added Hotel and Country Class

timchang27 committed:
Update timchang27.md

chintaiann committed:
Update DeveloperGuide.md

timchang27 committed:
Add and View Hotels Command

timchang27 committed:
Merge pull request #97 from timchang27/master

Add and View Hotels Command

chintaiann committed:
Update User Guide 

corrected command summary
FAQ

timchang27 committed:
Update DeveloperGuide.md

chintaiann committed:
Update README.md

chintaiann committed:
Update README.md

chintaiann committed:
Update README.md

timchang27 committed:
Update Help Command with Hotel related commands

timchang27 committed:
Merge pull request #98 from timchang27/master

Update Help Command with Hotel related commands

chintaiann committed:
Update DeveloperGuide.md

chintaiann committed:
implement sort and print package function

chintaiann committed:
Merge branch 'master' of https://github.com/chintaiann/tp

chintaiann committed:
Merge branch 'master' into master

chintaiann committed:
Merge pull request #1 from AY2122S2-CS2113-F10-3/master

pr1

chintaiann committed:
Merge pull request #99 from chintaiann/master

Merge pull request #1 from AY2122S2-CS2113-F10-3/master

chintaiann committed:
Update DeveloperGuide.md

chintaiann committed:
Update DeveloperGuide.md

chintaiann committed:
Update DeveloperGuide.md

chintaiann committed:
Merge branch 'AY2122S2-CS2113-F10-3:master' into master

chintaiann committed:
Update UserGuide.md

chintaiann committed:
Update data.txt

chintaiann committed:
Merge branch 'AY2122S2-CS2113-F10-3:master' into master

chintaiann committed:
Merge pull request #100 from chintaiann/master

update user guide and developer guide

chintaiann committed:
Update chintaiann.md

chintaiann committed:
Update AboutUs.md

timchang27 committed:
Update AboutUs.md

mafpovbul committed:
Update AboutUs.md

mafpovbul committed:
Update AboutUs.md

timchang27 committed:
Fixed AddHotelCommand

timchang27 committed:
Fixed AddHotelCommand

timchang27 committed:
Merge pull request #101 from timchang27/master

Fixed AddHotelCommand

Commit:
Add portfolio to AboutUs.md

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge branch 'master' into master

Commit:
Update AboutUs.md

Commit:
Update AboutUs.md

timchang27 committed:
Fixed AddHotelCommand

timchang27 committed:
Merge pull request #103 from timchang27/master

Removed some unused commands

timchang27 committed:
Update timchang27.md

Commit:
Merge branch 'master' into master

Commit:
Merge pull request #102 from edemirkirkan/master

Update

Commit:
Merge pull request #2 from AY2122S2-CS2113-F10-3/master

Merge pull request #102 from edemirkirkan/master

Commit:
Review bug issues

Commit:
Merge pull request #104 from edemirkirkan/master

Review bug issues
